astropy/coordinates/erfa_astrom.py

```diff
@@ -0,0 +1,269 @@
+                ' precision to about a microsecond of precision',
+                AstropyWarning
+            )
+        self.mjd_resolution = time_resolution.to_value(u.day)
+
+    def _get_support_points(self, obstime):
+        '''
+        Calculate support points for the interpolation.
+
+        We divide the MJD by the time resolution (as single float64 values),
+        and calculate ceil and floor.
+        Then we take the unique and sorted values and scale back to MJD.
+        This will create a sparse support for non-regular input obstimes.
+        '''
+        mjd_scaled = np.ravel(obstime.mjd / self. mjd_resolution)
+
+        # ndmin=1 needed in case only a single coordinate is transformed
+        mjd_lower = np.array(np.floor(mjd_scaled), ndmin=1, copy=False)
+        mjd_upper = np.array(np.ceil(mjd_scaled), ndmin=1, copy=False)
+
+        # unique already does sorting
+        mjd_u = np.unique(np.concatenate([mjd_lower, mjd_upper]))
+
+        return Time(
+            mjd_u * self.mjd_resolution,
+            format='mjd',
+            scale=obstime.scale,
+        )
+
+    @staticmethod
+    def _prepare_earth_position_vel(support, obstime):
+        pv_support, heliocentric_support = prepare_earth_position_vel(support)
+
+        # do interpolation
+        earth_pv = np.empty(obstime.shape, dtype=erfa.dt_pv)
+        earth_heliocentric = np.empty(obstime.shape + (3,))
+        for dim in range(3):
+            for key in 'pv':
+                earth_pv[key][..., dim] = np.interp(
+                    obstime.mjd,
+                    support.mjd,
+                    pv_support[key][..., dim]
+                )
+            earth_heliocentric[..., dim] = np.interp(
+                obstime.mjd, support.mjd, heliocentric_support[..., dim]
+            )
+
+        return earth_pv, earth_heliocentric
+
+    @staticmethod
+    def _get_cip(support, obstime):
+        jd1_tt_support, jd2_tt_support = get_jd12(support, 'tt')
+        cip_support = get_cip(jd1_tt_support, jd2_tt_support)
+        return tuple(
+            np.interp(obstime.mjd, support.mjd, cip_component)
+            for cip_component in cip_support
+        )
+
+    def apci(self, frame_or_coord):
+        '''
+        Wrapper for ``erfa.apci``, used in conversions CIRS <-> ICRS
+
+        Arguments
+        ---------
+        frame_or_coord: ``astropy.coordinates.BaseCoordinateFrame`` or ``astropy.coordinates.SkyCoord``
+            Frame or coordinate instance in the corresponding frame
+            for which to calculate the calculate the astrom values.
+            For this function, a CIRS frame is expected.
+        '''
+        obstime = frame_or_coord.obstime
+        # no point in interpolating for a single value
+        if obstime.size == 1:
+            return super().apci(frame_or_coord)
+
+        support = self._get_support_points(obstime)
+
+        jd1_tt, jd2_tt = get_jd12(obstime, 'tt')
```

Comment:
Total nitpick, but I think this would be clearer just before the `erfa.apci` call (indeed, could move it inside the call with `*get_jd12(obstime, 'tt')`)

Alternatively, moving it *before* the `support = self....` line would also work to make clearer what belongs together.

Reply:
I actually already had this change :)